dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIChatCompletionService.cs

```diff
@@ -0,0 +1,211 @@
+    {
+        var result = new StringBuilder();
+
+        await foreach (var content in this.RunInferenceAsync(chatHistory, executionSettings, cancellationToken).ConfigureAwait(false))
+        {
+            result.Append(content);
+        }
+
+        return new List<ChatMessageContent>
+        {
+            new(
+                role: AuthorRole.Assistant,
+                modelId: this._modelId,
+                content: result.ToString())
+        };
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
+    {
+        await foreach (var content in this.RunInferenceAsync(chatHistory, executionSettings, cancellationToken).ConfigureAwait(false))
+        {
+            yield return new StreamingChatMessageContent(AuthorRole.Assistant, content, modelId: this._modelId);
+        }
+    }
+
+    private async IAsyncEnumerable<string> RunInferenceAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings, CancellationToken cancellationToken)
+    {
+        OnnxRuntimeGenAIPromptExecutionSettings onnxRuntimeGenAIPromptExecutionSettings = OnnxRuntimeGenAIPromptExecutionSettings.FromExecutionSettings(executionSettings);
+
+        var prompt = this.GetPrompt(chatHistory, onnxRuntimeGenAIPromptExecutionSettings);
+        var tokens = this.GetTokenizer().Encode(prompt);
+
+        var generatorParams = new GeneratorParams(this.GetModel());
+        this.UpdateGeneratorParamsFromPromptExecutionSettings(generatorParams, onnxRuntimeGenAIPromptExecutionSettings);
+        generatorParams.SetInputSequences(tokens);
+
+        var generator = new Generator(this.GetModel(), generatorParams);
+
+        while (!generator.IsDone())
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            yield return await Task.Run(() =>
+            {
+                generator.ComputeLogits();
+                generator.GenerateNextToken();
+
+                var outputTokens = generator.GetSequence(0);
+                var newToken = outputTokens.Slice(outputTokens.Length - 1, 1);
+                var output = this.GetTokenizer().Decode(newToken);
+                return output;
+            }, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    private Model GetModel()
+    {
+        if (this._model == null)
+        {
+            this._model = new Model(this._modelPath);
+        }
+
+        return this._model;
+    }
+
+    private Tokenizer GetTokenizer()
+    {
+        if (this._tokenizer == null)
+        {
+            this._tokenizer = new Tokenizer(this.GetModel());
+        }
+
+        return this._tokenizer;
+    }
```

Comment:
```suggestion
    [MemberNotNull(nameof(_tokenizer))]
    private Tokenizer GetTokenizer() => this._tokenizer ??= new Tokenizer(this.GetModel());
```
Same here, consider making this a property.